docs/user_guide.md

```diff
@@ -719,6 +721,70 @@ If you are a project admin, you can disable a Robot Account by clicking "Disable
+
+|Event|Webhook Event Type|Contents of Notification|
+|---|---|---|
+|Push manifest to registry|`IMAGE PUSH`|Repository namespace name, repository name, tags, manifest digest, image name, image size, push time timestamp, username of user who pushed image|
+|Pull manifest from registry|`IMAGE PULL`|Repository namespace name, repository name, manifest digest, image name, pull time timestamp, username of user who pulled image|
+|Delete manifest from registry|`IMAGE DELETE`|Repository namespace name, repository name, manifest digest, image name, image size, delete time timestamp, username of user who deleted image|
+|Upload Helm chart to registry|`CHART PUSH`|Repository name, chart name, chart type, chart version, chart size, tag, timestamp of push, username of user who uploaded chart|
+|Download Helm chart from registry|`CHART PULL`|Repository name, chart name, chart type, chart version, chart size, tag, timestamp of push, username of user who pulled chart|
+|Delete Helm chart from registry|`CHART DELETE`|Repository name, chart name, chart type, chart version, chart size, tag, timestamp of delete, username of user who deleted chart|
+|Image scan completed|`IMAGE SCAN COMPLETED`|Repository namespace name, repository name, tag scanned, image name, number of critical issues, number of major issues, number of minor issues, last scan status, scan completion time timestamp, vulnerability information (CVE ID, description, link to CVE, criticality, URL for any fix), username of user who performed scan|
+|Image scan failed|`IMAGE SCAN FAILED`|Repository namespace name, repository name, tag scanned, image name, error that occurred, username of user who performed scan|
+
+### Webhook Endpoint Recommendations
+
+The endpoint that receives the webhook should ideally meet the following requirements:
+
+- Have a webhook listener that is capable of interpreting the payload and acting upon the information it contains. For example, running a shell script.
+- Support queues and retry configurations, so that failed webhook notifications are queued and retried until the request succeeds or number of retries exceeds the configured maximum.
+
+It is recommended to create a separate endpoint URL for each project. 
+
+### Example Use Cases
+
+You can configure your continuous integration and development infrastructure so that it performs the following types of operations when it receives a webhook notification from Harbor.
+
+- Image push: 
+  - Trigger a new build immediately following a push on selected repositories s or tags.
+  - Notify services or applications that use the image that a new image is available and pull it.
+  - Scan the image using Clair.
+  - Replicate the image to remote registries.
+- Image scanning:
+  - If a vulnerability is found, rescan the image or replicate it to another registry.
+  - If the scan passes, build the image.

```

Comment:
i think you mean deploy the image when the scan passes.

Reply:
Fixed in https://github.com/goharbor/harbor/pull/8869/commits/e1103b753e11b4d6a794d752620d84c3cccff54f.